src/Discord/Builders/Components/Section.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is a part of the DiscordPHP project.
+ *
+ * Copyright (c) 2015-present David Cole <david.cole1340@gmail.com>
+ *
+ * This file is subject to the MIT license that is bundled
+ * with this source code in the LICENSE.md file.
+ */
+
+namespace Discord\Builders\Components;
+
+/**
+ * Section components allow you to define up to 3 text display components and add either a thumbnail or button to the right side.
+ *
+ * @link https://discord.com/developers/docs/interactions/message-components#section
+ *
+ * @since 10.4.0
+ */
+class Section extends Component implements Contracts\ComponentV2
+{
+    /**
+     * Array of text display components.
+     *
+     * @var TextDisplay[]
+     */
+    private $components = [];
+
+    /**
+     * Accessory component (Thumbnail or Button).
+     *
+     * @var Thumbnail|Button|null
+     */
+    private $accessory;
+
+    /**
+     * Creates a new section.
+     *
+     * @return self
+     */
+    public static function new(): self
+    {
+        return new self();
+    }
+
+    /**
+     * Adds a text display component to the section.
+     * Text displays can only be used within sections.
+     *
+     * @param Component $component Text display component to add.
+     *
+     * @throws \InvalidArgumentException Component is not a TextDisplay.
+     * @throws \OverflowException Section exceeds 3 text components.
+     *
+     * @return $this
+     */
+    public function addComponent(Component $component): self
+    {
+        if (! ($component instanceof TextDisplay)) {
+            throw new \InvalidArgumentException('Section can only contain TextDisplay components. Use setAccessory() for Thumbnail or Button.');
```

Comment:
```suggestion
            throw new \InvalidArgumentException('Section can only contain TextDisplay components.');
```

You are not checking wether the `$component` is a `Thumbnail` or `Button` here so it's better not to explicitly say that.

Reply:
On a second though, the spec is telling us to not hardcode this, so we should not validate this yet
```suggestion
            trigger_error('Section may only contain TextDisplay components.', E_USER_NOTICE);
```

Reply:
Then don't throw the errors, remove the argument checking, and leave the type `Component` in the parameter while leaving the phpdoc `TextDisplay` for now